### DISCUSSION TITLE
Unable to install osxphotos using pipx on my Intel MacBook running macOS 13.5.2

### DISCUSSION THREAD

**Discussion by 420gofOGKush** [RhetTbull/osxphotos#1204] [Q&A] [answered]
Stumbled across this tool to backup my library with metadata since Image Capture wasn't doing the job. 
But while installing on terminal (pipx install osxphotos) I get an error thrown at me saying 

`pip failed to build package:
    xattr

Some possibly relevant errors from pip install:
    error: subprocess-exited-with-error
    xcrun: error: invalid active developer path (/Library/Developer/CommandLineTools), missing xcrun at: /Library/Developer/CommandLineTools/usr/bin/xcrun
    error: command '/usr/bin/clang' failed with exit code 1

Error installing osxphotos.`

I am not the most well versed person on the terminal side of things so I am not aware of what the error is talking about. Any help with install would be appreciated!
I am running 13.5.2 Ventura on my Intel 2019 MBP for additional information.

**Comment by RhetTbull** [comment]
You're the second person to report this (See #1177). The issue has to do with the `xattr` library which is a dependency of osxphotos. This package must be built from source and that requires installing the Xcode command line tools. However, that shouldn't be necessary as pre-built binaries are usually published and `pipx` will use those.  In this case, pipx isn't finding the right binary.  I'll open an issue with the `xattr` maintainer. In the mean time you should be able to do this:

Download [this](https://files.pythonhosted.org/packages/f8/9a/834e987c3545e8408a7dec183d42385d34e76d151e5ee3d8751b504a8e91/xattr-0.10.1-cp311-cp311-macosx_10_9_x86_64.whl) file (the prebuilt binary "wheel" package for xattr) then run `python3 -m pip install /path/to/xattr-0.10.1-cp311-cp311-macosx_10_9_x86_64.whl`

**Reply by RhetTbull** [reply]
Alternatively, you could [download the pre-built executable](https://github.com/RhetTbull/osxphotos/releases/download/v0.63.0/osxphotos_MacOS_exe_darwin_x86_64_v0.63.0.zip) and put it somewhere in your path (like `/usr/local/bin`)

**Reply by 420gofOGKush** [reply]
Hello! I was a little busy and didn't test again until today but after installing the prebuilt binary wheel thing, osxphotos seems to have installed but now when I try to run it I get this error

[4398] Module object for pyimod02_importers is NULL!
Traceback (most recent call last):
  File "PyInstaller/loader/pyimod02_importers.py", line 22, in <module>
  File "pathlib.py", line 14, in <module>
  File "urllib/parse.py", line 40, in <module>
ModuleNotFoundError: No module named 'ipaddress'
Traceback (most recent call last):
  File "PyInstaller/loader/pyiboot01_bootstrap.py", line 17, in <module>
ModuleNotFoundError: No module named 'pyimod02_importers'
[4398] Failed to execute script 'pyiboot01_bootstrap' due to unhandled exception!

What does this mean and what can I do? I also tried to use the prebuilt exe but that gives me the same error
Python version I am running is Python 3.11.5 (main, Aug 24 2023, 15:18:16) [Clang 14.0.3 (clang-1403.0.22.14.1)] on darwin

**Reply by RhetTbull** [reply]
I just tried the executable and got the same error (Intel 13.5.1) with the binary executable. Interestingly, on my M1 with the arm executable, it runs as expected. I'm not sure what the issue is with the executable as that has been working for a long time and I've not changed the build process. I'll open a separate issue for this. 

I'm not sure what to do about the xattr wheel issue. One idea might be to install the XCode command line tools using: `xcode-select --install` then try `brew install osxphotos` (or `brew reinstall osxphotos` if the install command complains).

Before doing that, maybe try the following:

```bash
brew uninstall pipx
brew uninstall python@3.11
brew install python@3.11
brew install pipx
brew install osxphotos
```

I'm still trying to figure out if this issue is with osxphotos, xattr, pipx, or pip or something else. It does work fine on many Intel Macs running Ventura including my 2019 Intel Macbook Air with 13.5.1.

**Reply by RhetTbull** [reply]
I updated pyinstaller and rebuilt the [x86x executable](https://github.com/RhetTbull/osxphotos/releases/download/v0.63.1/osxphotos_MacOS_exe_darwin_x86_64_v0.63.1.zip) so you could also try that again.

**Reply by RhetTbull** [reply]
I think I figured out the issue with xattr!  I am working on a fix. The version of xattr osxphotos was trying to use doesn't have a wheel built for python 3.11 so pip (and pipx) try to build it from source which fails without the Xcode tools installed. The fix is to update the version of xattr osxphotos uses.

**Reply by RhetTbull** [reply, marked as answer]
@420gofOGKush I think I've found and fixed the issue. Please give it another try with pipx. v0.63.3 fixes the dependencies that were causing pipx to try building xattr unnecessarily.

**Reply by 420gofOGKush** [reply]
Alright I'll try again! Sorry for the late response been a little busy. 
I'll get back to you with updates!

**Reply by 420gofOGKush** [reply]
Got the program working and I'll set up everything I need soon, thanks @RhetTbull for the fix!
